### DISCUSSION TITLE
why pflag are overriden by configuration file ?

### DISCUSSION THREAD

**Discussion by mortenko** [spf13/viper#1868] [Q&A] [answered]
Hi,
I'm struggling with the problem that **pflags** are overriden by configuration file.  

According the doc, the precedence is following: 

explicit call to Set
flag
env
config
key/value store
default

So I expect that config will not override flags.

Here is an example:

```
folder/config.yaml:
foo:
  foo2
```

```
`      // PFLAGS
	pflag.String("foo", "Foo-Override-Sample", "the foo it be")
	viper.BindPFlags(pflag.CommandLine)
	fmt.Println("foo is", viper.Get("foo")) // the result will be FOO-Override-Sample

	// CONFIG file 
	viper.SetConfigType("yaml")
	viper.SetConfigName("config") 
	viper.AddConfigPath("/folder/")
	err := viper.ReadInConfig()
	if err != nil {
		fmt.Println("No configuration file loaded.")
	}

        pflag.Parse()

	fmt.Println(viper.Get("foo"))` // the result will be foo2
```

At the end the result will be the value from configuration file. Why ? 

Thanks for answer

**Comment by sagikazarmark** [comment]
File values will override flag defaults. Setting the flag should take precedence though.

**Reply by mortenko** [reply]
@sagikazarmark little bit confused. How Could I set the flag that it will take precedence ?

**Reply by mortenko** [reply]
Do you mean 	pflag.Set("foo", "Foo-Override-Sample") ?

**Reply by sagikazarmark** [reply, marked as answer]
`./myapp --foo bar`?
